### PR TITLE
Invoke trust system in TF for Staged_ledger errors

### PR DESCRIPTION
**Pull request by psteckler**
`Transition_frontier.Breadcrumb.build` takes a trust system and an optional sender argument. If there's a sender, invoke the trust system when there's a staged ledger error.

In this PR, except for the error `Unexpected`, where there's no trust system invocation, the `Violated_protocol` action is used. We could match against particular errors, and do something finer.

- [ ] Tests were added for the new behavior
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them:

### CODE REVIEW

File: src/lib/transition_frontier/transition_frontier.ml [hunk 1]
```diff
@@ -97,18 +97,37 @@ struct
...
+                      let error_string =
+                        Staged_ledger.Staged_ledger_error.to_string
+                          staged_ledger_error
+                      in
+                      Trust_system.(
+                        record trust_system logger inet_addr
+                          Actions.
+                            ( Violated_protocol
+                            , Some
+                                ( "Staged_ledger error: $error"
```
Comment (deepthiskumar) [inline src/lib/transition_frontier/transition_frontier.ml]: This failure is recorded twice, first here as `Violated_protocol` and then as `Gossiped_invalid_transition` in `processor.ml`. Shouldn't it be recorded just once?

**Comment by psteckler** [comment]
@enolan should we have different actions for the different staged ledger errors, and remove the catchall `Gossiped_invalid_transition` action?

File: src/lib/transition_frontier/transition_frontier.ml (hunk 1 above)
Comment (psteckler) [reply src/lib/transition_frontier/transition_frontier.ml]: removed this use of the trust system

distinguished errors in build

**Comment by enolan** [comment]
I'd like to reuse `Sent_invalid_signature` from my other PR and add a generic `Sent_invalid_proof`, to be used here and in the snark pool code. The rest can stay the same

**Comment by deepthiskumar** [comment]
https://github.com/CodaProtocol/coda/blob/f82883525d294c49886468ade45ad5ba9c9fc5ff/src/lib/transition_frontier/transition_frontier.ml#L157
Record this error as well since we no longer do it in `processor.ml`

**Pull request merged**